spec/controllers/concerns/ial2_profile_concern_spec.rb

```diff
@@ -0,0 +1,209 @@
+require 'rails_helper'
+
+RSpec.describe Ial2ProfileConcern do
+  let(:test_controller) do
+    Data.define(
+      :current_user,
+      :user_session,
+      :analytics,
+    ) do
+      include Ial2ProfileConcern
+    end
+  end
+
+  let(:user) { create(:user) }
+  let(:user_session) { {} }
+  let(:analytics) { double(Analytics) }
+  let(:encryption_error) { Encryption::EncryptionError.new }
+  let(:cacher) { double(Pii::Cacher) }
+  let(:password) { 'PraiseTheSun!' }
+
+  describe '#cache_profiles' do
+    subject { test_controller.new(current_user: user, user_session:, analytics:) }
+
+    context 'when the user has a pending profile' do
+      let!(:profile) { create(:profile, :in_person_verification_pending, user:) }
+
+      context 'when the profile can be saved to cache' do
+        before do
+          allow(cacher).to receive(:save)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          subject.cache_profiles(password)
+        end
+
+        it 'stores the decrypted profile in cache' do
+          expect(cacher).to have_received(:save).with(password, profile)
+        end
+      end
+
+      context 'when the profile can not be saved to cache' do
+        before do
+          allow(cacher).to receive(:save).and_raise(encryption_error)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          allow(analytics).to receive(:profile_encryption_invalid)
+          subject.cache_profiles(password)
+        end
+
+        it 'deactivates the profile with reason encryption_error' do
+          expect(profile.reload).to have_attributes(
+            active: false,
+            deactivation_reason: 'encryption_error',
```

Comment:
Should we also include `in_person_verification_pending_at`?

Reply:
I am no longer setting `in_person_verification_pending_at` to nil from the deactivate_due_to_encryption_error any more based on [Matt's review comment](https://github.com/18F/identity-idp/pull/11585/files/153aee87b9454943008ccc35e821d575e16c9c9c#r1868482292).

Reply:
Ahh, yes. Resolving now